### DISCUSSION TITLE
After indexing, during compaction, "too many open files" macOS 12.1 Monterey

### DISCUSSION THREAD

**Discussion by grgtog** [romanz/electrs#663] [Q&A] [answered]
Upon first running of elects, indexing goes to completion. Then compaction begins. After several minutes the following error occurs:

[2022-01-13T02:23:00.414Z INFO  electrs::db] starting config compaction
[2022-01-13T02:23:00.421Z INFO  electrs::db] starting headers compaction
[2022-01-13T02:23:01.841Z INFO  electrs::db] starting txid compaction
[2022-01-13T02:29:28.203Z INFO  electrs::db] starting funding compaction
[2022-01-13T02:29:29.706Z INFO  electrs::db] starting spending compaction
thread 'main' panicked at 'DB::put failed: Error { message: "IO error: While open a file for random read: /Volumes/T7/ElectrsDB/bitcoin/000083.sst: Too many open files" }', src/db.rs:340:14

I have increased the file limits using:
sudo sysctl -w kern.maxfiles=20480
and
sudo sysctl -w kern.maxfilesperproc=24576

then run the following to verify:
sysctl -A | grep kern.maxfiles
kern.maxfiles: 20480
kern.maxfilesperproc: 24576

Run electro again, same problem "Too many open files"

Any suggestions?
thanks...

**Comment by grgtog** [comment]
Sorry for the typos, should of course be electrs, not elects or electro.

**Comment by torkelrogstad** [comment, marked as answer]
I got around this by increasing the file limit with `❯ ulimit --file-descriptor-count 10000` (the number is chosen very arbitrarily). Note that this is only for the currently running process, so it won't persist across terminal windows for example
